projects/admin/src/app/record/search-view/document-advanced-search-form/advanced-search.service.spec.ts

```diff
@@ -0,0 +1,228 @@
+/*
+ * RERO ILS UI
+ * Copyright (C) 2019-2023 RERO
+ * Copyright (C) 2021-2023 UCLouvain
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, version 3 of the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+import { TestBed } from '@angular/core/testing';
+import { DocumentApiService } from '@app/admin/api/document-api.service';
+import { cloneDeep } from 'lodash-es';
+import { of } from 'rxjs';
+import { AdvancedSearchService } from './advanced-search.service';
+
+describe('AdvancedSearchService', () => {
+  let service: AdvancedSearchService;
+
+  const apiResponse = {
+    fieldsConfig: [
+      {
+        field: "title.*",
+        label: "Title",
+        value: "title",
+        options: {
+          search_type: [
+            { label: "contains", value: AdvancedSearchService.SEARCH_TYPE_CONTAINS },
+            { label: "exact", value: AdvancedSearchService.SEARCH_TYPE_CONTAINS_PHRASE },
```

Comment:
"exact" is misleading. Please replace by "contains phrase".

Reply:
It's important to keep the wording short, otherwise it's beyond select. That's why I chose `exact`.

Reply:
It doesn't matter here, because it's a test, but I can change it when we've defined a new short chain.